pkg/aws/aws.go

```diff
@@ -124,65 +124,98 @@ type AwsProvider struct {
 	WorkingDirectory string
 }
 
-func GetSubnetID(ctx context.Context, provider *AwsProvider) (string, error) {
-	svc := ec2.NewFromConfig(provider.AwsConfig)
-
-	// first search for a default devpod specific subnet, if it fails
-	// we search the subnet with most free IPs that can do also public-ipv4
-	input := &ec2.DescribeSubnetsInput{
-		Filters: []types.Filter{
-			{
-				Name: aws.String("tag:devpod"),
-				Values: []string{
-					"devpod",
-				},
-			},
-		},
+func GetSubnet(ctx context.Context, provider *AwsProvider) (string, error) {
+	// in case a single subnet ID is specified, use it without further checks
+	if len(provider.Config.SubnetIDs) == 1 {
+		return provider.Config.SubnetIDs[0], nil
 	}
 
-	result, err := svc.DescribeSubnets(ctx, input)
-	if err != nil {
-		return "", err
-	}
+	svc := ec2.NewFromConfig(provider.AwsConfig)
+	// in case multiple subnet IDs are specified, we return the one with most free IPs
+	if len(provider.Config.SubnetIDs) > 1 {
+		subnets, err := svc.DescribeSubnets(ctx, &ec2.DescribeSubnetsInput{
+			SubnetIds: provider.Config.SubnetIDs,
+		})
+		if err != nil {
+			return "", fmt.Errorf("list specified subnets %q: %w", provider.Config.SubnetIDs, err)
+		}
+		var maxIPCount int32
+		var subnet *types.Subnet
+		for _, s := range subnets.Subnets {
+			if provider.Config.AvailabilityZone != "" && *s.AvailabilityZone != provider.Config.AvailabilityZone {
```

Comment:
what's the need to dereference `s` here?

Reply:
Hi Bryan, I'm not sure I understand this question correctly. Im dereferencing the field `AvailabilityZone`, which is a pointer. Otherwise the compiler would complain:
![image](https://github.com/user-attachments/assets/a48aa66c-1050-4c3d-8c51-a1e7cb38e605)